internal/consensus/state.go

```diff
@@ -272,6 +274,24 @@ func (cs *State) SetPrivValidator(priv types.PrivValidator) {
 
 	cs.privValidator = priv
 
+	if priv != nil {
+		switch t := priv.(type) {
+		case *privval.RetrySignerClient:
+			cs.privValidatorType = types.RetrySignerClient
+		case *privval.FilePV:
+			cs.privValidatorType = types.FileSignerClient
+		case *privval.SignerClient:
+			cs.privValidatorType = types.SignerClient
+		case types.MockPV:
+			cs.privValidatorType = types.MockSignerClient
+		case *types.ErroringMockPV:
+			cs.privValidatorType = types.ErrorMockSignerClient
+		default:
+			cs.Logger.Error("unsupported priv validator type", "err",
+				fmt.Errorf("error privValidatorType %s", t))
+		}
+	}
```

Comment:
I'm getting this error in the e2e tests:
```
validator01    | 2021-07-01T10:34:03Z ERROR unsupported priv validator type err="error privValidatorType &{{{{%!s(*log.syncWriter=&{{0 0} {0xc00018e010 true 2006-01-02T15:04:05Z07:00 [] [] <nil> 0xb98500 <nil> <nil> <nil> <nil> <nil> <nil>}})} %!s(zerolog.Level=1) <nil> {\"module\":\"main\" [{}] %!s(bool=false)} %!s(bool=false)} %!s(*privval.privValidatorAPIClient=&{0xc000240000}) %!s(*grpc.ClientConn=<nil>) gen-0008}" module=consensus
```
Did we miss a case? Seems to be the grpc priv val

Reply:
@cmwaters thanks for catching, I will add the missing type today.

Reply:
@cmwaters how do I run e2e tests to see this error?
I want to verify my fixes on it.
so I did but cannot see the error:
1. make
2. ./build/runner -f networks/simple.toml start
3. ./build/runner -f networks/simple.toml test

Reply:
Just run 
./build/runner -f networks/ci.toml 
Without any other command and it should do the complete e2e test process. You'll need to change the validators to all use the GRPC priv Val

Reply:
@cmwaters submitted PR #6645